### ISSUE TITLE
Transform throwing "Not a legal OleAut date" exception

### ISSUE THREAD

**Issue by JPhilC** [ASCOMInitiative/ASCOMPlatform#5]
Peter,

I originially posted this under the NOVAS3.1 error with the same exception but have created a new issue as I think it may be different. To help you track the problem down here is a little unit test to replicate the problem.

```c-sharp
using System;
using ASCOM.Astrometry.Transform;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace LunaticAstroEQ.Tests
{
   [TestClass]
   public class TransformTest
   {
      [TestMethod]
      public void TestMethod1()
      {
         using (Transform transform = new Transform())
         {
            double longitude = 1.3223722222222223;
            transform.SiteLatitude = 52.65177222222222;
            transform.SiteLongitude = longitude;
            transform.SiteElevation = 175.5;
            transform.JulianDateUTC = LocalApparentSiderealTime(longitude, new DateTime(2017, 03, 28, 21, 0, 0));
            transform.SetTopocentric(0.13166666666666665, 29.0);
            double alt = transform.ElevationTopocentric;
            double az = transform.AzimuthTopocentric;
            Assert.AreEqual(alt, 10.0); 
         }
      }

      private double LocalApparentSiderealTime(double longitude, DateTime localTime)
      {
         // get greenwich sidereal time: https://en.wikipedia.org/wiki/Sidereal_time
         //double siderealTime = (18.697374558 + 24.065709824419081 * (utilities.DateUTCToJulian(DateTime.UtcNow) - 2451545.0));

         // alternative using NOVAS 3.1
         double siderealTime = 0.0;
         using (var novas = new ASCOM.Astrometry.NOVAS.NOVAS31())
         {
            var jd = localTime.ToUniversalTime().ToOADate() + 2415018.5;      // Taken from ASCOM.Util.DateUTCToJulian
            novas.SiderealTime(jd, 0, novas.DeltaT(jd),
                ASCOM.Astrometry.GstType.GreenwichApparentSiderealTime,
                ASCOM.Astrometry.Method.EquinoxBased,
                ASCOM.Astrometry.Accuracy.Reduced, ref siderealTime);
         }
         // allow for the longitude
         siderealTime += longitude / 360.0 * 24.0;
         // reduce to the range 0 to 24 hours
         siderealTime = siderealTime % 24.0;
         return siderealTime;
      }
   }
}

```
The exeption will be thrown on the line `double alt = transform.ElevationTopocentric;`

Looking at the exception call stack it is the following line that is causing the problem.
`                LogDebugMessage("DeltaUT1(JD)", String.Format("Automatic DeltaUT1 is required for Julian date: {0} ({1})", RequiredDeltaUT1JulianDateUTC,
                                                       DateTime.FromOADate(RequiredDeltaUT1JulianDateUTC - OLE_AUTOMATION_JULIAN_DATE_OFFSET).ToString(DOWNLOAD_TASK_TIME_FORMAT)))
`
Line 649 in EarthRotationParameters.vb. I think you have an error in your debug message logging.

The call stack is:
```
   at System.DateTime.DoubleDateToTicks(Double value)
   at System.DateTime.FromOADate(Double d)
   at ASCOM.Astrometry.EarthRotationParameters.DeltaUT1(Double RequiredDeltaUT1JulianDateUTC) in C:\ASCOM Build\Export\ASCOM.Astrometry\ASCOM.Astrometry\EarthRotationParameters.vb:line 649
   at ASCOM.Astrometry.AstroUtils.AstroUtils.DeltaUT(Double JulianDate) in C:\ASCOM Build\Export\ASCOM.Astrometry\ASCOM.Astrometry\AstroUtils.vb:line 400
   at ASCOM.Astrometry.Transform.Transform.TopoToJ2000() in C:\ASCOM Build\Export\ASCOM.Astrometry\ASCOM.Astrometry\Transform.vb:line 582
   at ASCOM.Astrometry.Transform.Transform.Recalculate() in C:\ASCOM Build\Export\ASCOM.Astrometry\ASCOM.Astrometry\Transform.vb:line 653
   at ASCOM.Astrometry.Transform.Transform.get_ElevationTopocentric() in C:\ASCOM Build\Export\ASCOM.Astrometry\ASCOM.Astrometry\Transform.vb:line 438
   at LunaticAstroEQ.Tests.TransformTest.TestMethod1() in C:\Users\phil\source\repos\LunaticAstroEQ\LunaticAstroEQ.Tests\UnitTest1.cs:line 21
```

Kind regards,

Phil

p.s. I tried forking the repository and referencing the sources directly from a test project to debug this further but it wouldn't run the sources, just the installed platform DLLs. I don't want to screw up my development machine so I have had to stop.

**Comment by Peter-Simpson** [comment]
Hi Phil,

Thanks for providing your test code, however, I don't think this is a Platform issue because you appear to be setting an incorrect value for Transform's JulianDateUtc parameter. 

The property is expecting a Julian Date value (about 2457841.375 for your test date) but you actually supply a local sidereal time value, in the range 0..24 hours, which is well outside ASCOM's valid Julian Date range - hence the exception is thrown when the incorrect value is first used.

Regards, Peter

**Comment by JPhilC** [comment]
Hi Peter,

I agree I was supplying an incorrect value. But, there is an error in the platform error handling code. It is the code trying to construct the meaningful error message that is throwing the exception. I would investigate further but as I said simply cloning the sources didn't work as my test code simply ran the release versions DLLs.

Phil

**Comment by Peter-Simpson** [comment]
Hi Phil,

Thanks, it seems that I fixed this back in January in commit 984ad0df08ff415a3774679171088b9a2ab0abf5 but didn't post here. I've added checks in several places to ensure that Julian dates outside the supported range are rejected.

Regards, Peter
